### PR TITLE
Bump Newton to commit 81cdcfc

### PR DESCRIPTION
**Pull request by mmichelis**
# Description

Bumps Newton from `c7ae7c7648cd0717df39e5c94b95d5a02c997320` to `81cdcfc2dd89f8b7285e32b5e3853092a97fa6f9` and pins compatible MuJoCo `3.10.0`, MuJoCo-Warp `3.10.0.2`, and Newton USD schemas `>=0.4.0`. Upstream comparison: [c7ae7c7...81cdcfc](https://github.com/newton-physics/newton/compare/c7ae7c7648cd0717df39e5c94b95d5a02c997320...81cdcfc2dd89f8b7285e32b5e3853092a97fa6f9).

Migrates the Newton Warp renderer to the current tiled-camera APIs. The contact regression applies a 0.01 m lateral offset only to the affected CUDA MuJoCo-Warp mesh-capsule case tracked by [google-deepmind/mujoco_warp#1527](https://github.com/google-deepmind/mujoco_warp/issues/1527).

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Screenshots

Not applicable.

## Validation

- 1,016 unique `isaaclab_newton` cases before the develop rebase; after the rebase, both focused hard-reset cases and all 80 manager cases passed
- Post-rebase contact suite: 83 passed and 6 xpassed
- Post-rebase Newton Warp renderer: 5 of 5 strict AOV cases
- Post-rebase RL entrypoints: 7 core tests and 2 RSL-RL typed preset tests
- Five post-rebase unified-CLI Newton training smokes across MJWarp and Kamino
- 11 upstream ViewerUSD tests
- Pin, installation, changelog, and repository-wide pre-commit checks

## Checklist

- [x] I have read and understood the [contribution guidelines](https://isaac-sim.github.io/IsaacLab/main/source/refs/contributing.html)
- [x] I have run the [pre-commit checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added a changelog fragment under `source/<pkg>/changelog.d/` for every touched package
- [x] I have added my name to `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

**Review by AntoineRichard — APPROVED** [review]

File: source/isaaclab_newton/test/sensors/test_contact_sensor.py [hunk 1]
```diff
@@ -224,6 +224,10 @@ def test_horizontal_collision_detects_contact(device: str, use_mujoco_contacts:
         sim._app_control_on_stop_handle = None
 
         max_separation = max(cfg[1] for cfg in group_configs)
+        # Avoid MuJoCo-Warp #1527's symmetric zero-margin GJK edge case.
+        lateral_offset = (
+            0.01 if use_mujoco_contacts and device.startswith("cuda") and shape_type == ShapeType.MESH_CAPSULE else 0.0
+        )
```
Comment (hujc7) [inline source/isaaclab_newton/test/sensors/test_contact_sensor.py:230]: This seems test only change. Do we need to adapt impl code?

**Review by huidongc — APPROVED** [review]

**Review by mataylor-nvidia — APPROVED** [review]

File: source/isaaclab_newton/test/sensors/test_contact_sensor.py (hunk 1 above)
Comment (AntoineRichard) [reply source/isaaclab_newton/test/sensors/test_contact_sensor.py:230]: @mmichelis

Comment (mmichelis) [reply source/isaaclab_newton/test/sensors/test_contact_sensor.py:230]: Thanks for bringing this up, indeed it would make sense if this is not just a test change. But I had an agent do a deep dive into all environments, and it seems dexsuite is the closest to this zero-margin edge case, but they use the newton collision pipeline and not mujoco contacts, so this bug is not triggered. It does indeed remain a test-case only change as far as I'm aware.
